packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -742,21 +746,43 @@ class _InterpolationSimulation extends Simulation {
 }
 
 class _RepeatingSimulation extends Simulation {
-  _RepeatingSimulation(this.min, this.max, Duration period)
-    : _periodInSeconds = period.inMicroseconds / Duration.microsecondsPerSecond {
+  _RepeatingSimulation(double initialValue, this.min, this.max, this.reverse, Duration period)
+    : _periodInSeconds = period.inMicroseconds / Duration.microsecondsPerSecond,
+      _initialT = (initialValue / (max - min)) * (period.inMicroseconds / Duration.microsecondsPerSecond) {
     assert(_periodInSeconds > 0.0);
+    assert(_initialT >= 0.0);
+
+    if (initialValue == max && reverse) {
+      _isPlayingReverse = true;
+    }
   }
 
   final double min;
   final double max;
+  final bool reverse;
 
   final double _periodInSeconds;
+  final double _initialT;
+
+  double _previousT = 0.0;
+  bool _isPlayingReverse = false;
 
   @override
   double x(double timeInSeconds) {
     assert(timeInSeconds >= 0.0);
-    final double t = (timeInSeconds / _periodInSeconds) % 1.0;
-    return ui.lerpDouble(min, max, t);
+
+    final double t = ((timeInSeconds + _initialT) / _periodInSeconds) % 1.0;
+
+    if (_previousT > t) {
```

Comment:
Shouldn't this be: `_isPlayingReverse = _previousT > t;` ?

Reply:
I have worked out a new and more robust solution for this and the `_previousT` was removed completely 😃 The new implementation is now independent from former ticks and does not need the old time anymore.